spec/index.bs

```diff
@@ -5239,36 +5281,95 @@ enum GPUStoreOp {
 
     : <dfn>drawIndirect(indirectBuffer, indirectOffset)</dfn>
     ::
+        Draws primitives using parameters read from a {{GPUBuffer}}.
+
+        The <dfn dfn>indirect draw parameters</dfn> encoded in the buffer must be a tightly
+        packed block of **four 32-bit unsigned integer values (16 bytes total)**, given in the same
+        order as the arguments for {{GPURenderEncoderBase/draw()}}. Written as a C struct,
+        the data layout would be:
+
+        ```c
+        struct GPUDrawIndirectParameters {
+            uint32_t vertexCount;
+            uint32_t instanceCount;
+            uint32_t firstVertex;
+            uint32_t firstInstance;
+        };
+        ```
 
         <div algorithm="GPURenderEncoderBase.drawIndirect">
             **Called on:** {{GPURenderEncoderBase}} this.
 
             **Arguments:**
             <pre class=argumentdef for="GPURenderEncoderBase/drawIndirect(indirectBuffer, indirectOffset)">
-                indirectBuffer:
-                indirectOffset:
+                |indirectBuffer|: Buffer containing the [=indirect draw parameters=].
+                |indirectOffset|: Offset in bytes into |indirectBuffer| where the drawing data begins.
             </pre>
 
             **Returns:** void
 
-            Issue: Describe {{GPURenderEncoderBase/drawIndirect()}} algorithm steps.
+            Issue the following steps on the [=Queue timeline=] of |this|:
+            <div class=queue-timeline>
+                1. If any of the following conditions are unsatisfied, generate a
+                    {{GPUValidationError}} error and stop.
+                    <div class=validusage>
+                        - |indirectBuffer| is [=valid=].
+                        - |indirectBuffer|.{{GPUObjectBase/[[device]]}} is |this|.
+                        - |indirectOffset| + sizeof([=indirect draw parameters=]) &le;
+                            |indirectBuffer|.{{GPUBuffer/[[size]]}}.
+
+                        Issue: Does |indirectOffset| need a particular alignment?
```

Comment:
Yes,  it needs to be aligned to 4 in Vulkan and Metal at least, likely in D3D12 as well